engine/runtime/src/main/java/org/enso/interpreter/node/expression/builtin/meta/EnsoProjectNode.java

```diff
@@ -0,0 +1,135 @@
+                      : "Should be called only from Enso code, and thus, should always have a root node";
+                  if (callRootNode instanceof EnsoRootNode ensoRootNode) {
+                    var pkg = ensoRootNode.getModuleScope().getModule().getPackage();
+                    // Don't return null, as that would signal to Truffle that we want to
+                    // continue the iteration.
+                    if (pkg != null) {
+                      return Optional.of(pkg);
+                    } else {
+                      return Optional.empty();
+                    }
+                  } else {
+                    throw new IllegalStateException(
+                        "Should not reach here: callRootNode = "
+                            + callRootNode
+                            + ". Probably not called from Enso?");
+                  }
+                },
+                // The first frame is always Enso_Project.enso_project
+                1);
+    if (pkgOpt.isPresent()) {
+      return createProjectDescriptionAtom(ctx, pkgOpt.get());
+    } else {
+      return notInModuleError(ctx);
+    }
+  }
+
+  @Specialization(guards = "!isNothing(interop, module)")
+  @TruffleBoundary
+  public Object getOtherProjectDescr(
+      Object module,
+      @CachedLibrary(limit = "5") InteropLibrary interop,
+      @CachedLibrary(limit = "5") TypesLibrary typesLib) {
+    var ctx = EnsoContext.get(this);
+    if (!typesLib.hasType(module)) {
+      return unsupportedArgsError(module);
+    }
+    Type moduleType = typesLib.getType(module);
+    // Currently, the module is represented as Type with no constructors.
+    if (!moduleType.getConstructors().isEmpty()) {
+      return unsupportedArgsError(module);
+    }
+    var pkg = moduleType.getDefinitionScope().getModule().getPackage();
+    if (pkg != null) {
+      return createProjectDescriptionAtom(ctx, pkg);
+    } else {
+      return notInModuleError(ctx);
+    }
+  }
+
+  private static Atom createProjectDescriptionAtom(EnsoContext ctx, Package<TruffleFile> pkg) {
+    EnsoFile rootPath = new EnsoFile(pkg.root().normalize());
+    Object cfg = ctx.getEnvironment().asGuestValue(pkg.config());
+    return ctx.getBuiltins()
+        .getProjectDescription()
+        .getUniqueConstructor()
+        .newInstance(rootPath, cfg);
+  }
+
+  private DataflowError unsupportedArgsError(Object moduleActual) {
+    return DataflowError.withoutTrace(
+        EnsoContext.get(this)
+            .getBuiltins()
+            .error()
+            .makeUnsupportedArgumentsError(
+                new Object[] {moduleActual}, "The `module` argument does not refer to a module"),
+        this);
+  }
+
+  private DataflowError notInModuleError(EnsoContext ctx) {
+    return DataflowError.withoutTrace(
+        ctx.getBuiltins().error().makeModuleNotInPackageError(), this);
+  }
+
+  boolean isNothing(InteropLibrary interop, Object object) {
+    return interop.isNull(object);
```

Comment:
TCA: It'd be more lightweight to check just `ctx.getBuiltins().nothing() == object`.

Reply:
@radeusgd is correct in this case. `InteropLibrary.isNull` is more general and, therefore, should deal even with these cases when called with a polyglot null.

Reply:
There is a cost of `InteropLibrary` (polymorphic, heavyweight, requiring too much `Node` instances). The only reason for `Nothing` is to support a _"default value"_. There is no point in supporting JavaScript `null` instead. Any value different from `Nothing|Module` may easily yield an error.

If we really want to support such foreign values, then we should have a specific `IsNullNode` or `IsNothingNode` and use it. It would specialize for `Nothing`, but could handle interop values as well, when they appear. Similar to #6299 idea.

Reply:
`InteropLibrary` overhead removed in e2a440d7f1b4bed284b0b5b45885524823044d9e. @radeusgd Note that if the wrong argument is passed, Unsupported_Arguments error is returned.